### PR TITLE
workaround for multithreading locale problem in GEOS

### PR DESCRIPTION
**Pull request by alex85k**
Found it here: https://msdn.microsoft.com/en-us/library/ms235302.aspx

### CODE REVIEW

File: osm2pgsql.cpp [hunk 1]
```diff
@@ -45,6 +46,7 @@
 
 int main(int argc, char *argv[])
 {
+    std::setlocale(LC_NUMERIC, "C");
```
Comment (alex85k) [inline osm2pgsql.cpp]: This may help to avoid problem on Linux with changed locale (did not help on Windows, however)

File: osmdata.cpp [hunk 2]
```diff
@@ -170,6 +170,10 @@ struct pending_threaded_processor : public middle_t::pending_processor {
     typedef std::pair<std::shared_ptr<const middle_query_t>, output_vec_t> clone_t;
 
     static void do_jobs(output_vec_t const& outputs, pending_queue_t& queue, size_t& ids_done, std::mutex& mutex, int append, bool ways) {
+#ifdef _MSC_VER
+	// Avoid problems when GEOS WKT-related methods switch the locale
+        _configthreadlocale(_ENABLE_PER_THREAD_LOCALE);
+#endif
         while (true) {
```
Comment (alex85k) [inline osmdata.cpp:177]: Is it the right place?

File: osm2pgsql.cpp (hunk 1 above)
Comment (lonvia) [inline osm2pgsql.cpp]: I'm not sure what other side effects a global change of the locale has. If it is running fine on Windows without this, I'd say we just go with the _configthreadlocale below.

Comment (alex85k) [inline osm2pgsql.cpp]: OK, removing this. It worked without it on Linux and will work, we hope.

**Pull request merged**